contractcourt/htlc_timeout_resolver.go

```diff
@@ -311,6 +370,7 @@ func (h *htlcTimeoutResolver) Resolve() (ContractResolver, error) {
 	// spend of the output, and make our next move off of that. Depending
 	// on if this is our commitment, or the remote party's commitment,
 	// we'll be watching a different outpoint and script.
+	// TODO(halseth): these details might not be correct after re-signing?
```

Comment:
Yeah in this case, the logic needs a large overhaul: we need to wait for the spend, of the output, then only then can we use that to register for confirmations below. Alternatively, capture the second value from `SweepInput` above and use that to listen to know once the output has been spent and confirmed on chain. 

So you can possibly abstract away the "output has been spent" logic here, to present the caller at this level with a single channel, that's either the sweeper ntfn channel or this raw spend notification. The logic _after_ the spend comes through should be identical (need to know if it was a timeout or a success spend).

Reply:
I think it should be correct now. The `chainDetailsToWatch` will stay the same, as the outpoint spent by the second level stays the same even though the sweeper re-signs it. Now we use that spend to determine the second-level output later.